### PR TITLE
Updated firm_capacity tutorial

### PR DESCRIPTION
**Pull request by OFR-IIASA**
The updated tutorial provides more explanatory text and matches the numbers in the text to that in the code.

# Please confirm that this PR has done the following:

- [x] Tests Added
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Adding to RELEASE_NOTES.md

Please add a single line in the release notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

**Comment by OFR-IIASA** [comment]
I have had this problem before with the CircleCI tests and unfortunately @gidden and I were unable to resolve this issue.

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 1]
```diff
@@ -1,5 +1,16 @@
 {
  "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Westeros Tutorial Part IVa. - Introducing firm capacity\n",
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: The term "firm capacity" is quite specific, so I would use it in quotes when first used. Also, I would make it clear that the description below is the explanation for that term.

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+    "\n",
+    "The next tutorial is comprised of three parts. In the first part we will show you how to ensure that in a model setup which doesnt use sub-annual timesteps and therefore does not depict peak load, we still account for sufficient backup capacity from electricity generation plants to maintain reliability through reasonable load and contingency events.\n",
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: *"... three parts. In the first part, ..."* - but there is no further reference to a second and third part?

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 3]
```diff
@@ -1,5 +1,16 @@
...
+    "\n",
+    "Further information can be found in https://doi.org/10.1016/j.esr.2013.01.001"
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Please also add a text-style citation like *(Sullivan et al., 2013)*

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 4]
```diff
@@ -40,29 +49,38 @@
...
     "model = 'Westeros Electrified'\n",
-    "scen = base.clone(model, 'firm_capacity','illustration of firm-capacity formulation', keep_sol=False)\n",
-    "scen.check_out()\n",
-    "\n",
-    "year_df = scen.vintage_and_active_years()\n",
-    "vintage_years, act_years = year_df['year_vtg'], year_df['year_act']\n",
-    "model_horizon = scen.set('year')\n",
-    "country = 'Westeros'"
+    "base = message_ix.Scenario(mp, model=model, scenario='emission_bound')\n",
+    "scen = base.clone(model, 'firm_capacity','illustration of firm-capacity formulation', keep_solution=False)\n",
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: pep8 - no whitespace after `,`

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 5]
```diff
@@ -40,29 +49,38 @@
...
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "### Load Factors - Describing the Electricity Sector\n",
-    "#### Peak load \n",
-    "If we further specify our demand, we will find that the demand for electricity (!not the demand for useful light but for electricity!) has a peak load that is different from the annual average load; e.g: if annual average load is 5 GWh peak load during midday might be 10GW --> the peak load factor is equal to 2."
+    "## Load Factors - Describing the Electricity Sector\n",
+    "### Peak load \n",
+    "For systems across the U.S., load duration curves are similar from one location to another. Based on load data it was calculated calculated that the peak load is, on average, 1.7 times the average load (a load factor of 59%).8 Coupled with a standard reserve margin of 15–20% of peak load, that means firm capacity requirements for the U.S. should be roughly twice average load.\n",
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Please make clear that this is the finding from Sullivan et al.

Also, the word "calculated" is duplicated, and the number "8" is a copy-paste error.

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 6]
```diff
@@ -40,29 +49,38 @@
...
+    "Therefore, a peak load factor of 2 is introduced for the level:secondary and commodity:electricity, which is level/commodity onto which all power generation technologies deliver."
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Please use some formatting for the level and commidity names. Also, "deliver" is jargon, maybe "where powerplant generation is accounted"?

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 7]
```diff
@@ -80,17 +98,20 @@
...
+    "## Rating bins and reliability factors - Describing the powerplant reliability\n",
+    "### Ensuring enough firm capacity\n",
+    "Toward meeting the firm capacity requirement, conventional generating technologies contribute their nameplate generation capacity. \n",
+    "\n",
+    "This therefore implies that for conventional powerplants, a reliability factor of 1 is assumed.\n",
+    "\n",
+    "In order to account for the integration challenge at higher shares of variable renewables, their contribution to the capacity requirements, capacity value, declines as the market share of the technology increases.  We therefore assume, that up to a share of 20% of total electricity supply, the capacity of wind_ppl adds by 80% to the firm capacity of the power system (r1). The remaining 80% installed capacity, however  only  5% contribute to the firm capcity (r2). "
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: suggest to replace *, capacity value,* by *(i.e., their capacity value)*

Please use formatting for names of technologies and rating bins, too.

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 8]
```diff
@@ -184,7 +205,38 @@
...
+   "metadata": {},
+   "source": [
+    "When comparing the results of the original scenario without the reliability factor ('emission_bound') and the results of our newly modified scenario ('firm_capacity'), for the same carbon price we can observe that the activity of the wind and the coal powerplant is identical, which is what we hoped to achieve."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Acitivity for scenario: 'emission_bound' (without firm capacity)"
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Typo in Activity

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 9]
```diff
@@ -184,7 +205,38 @@
...
-    "p = Plots(scen, country, firstyear=700)"
+    "p = Plots(scen, country, firstyear=700) # scenario: 'firm_capacity' (emission_bound scenario with firm capacity)\n",
+    "b = Plots(base, country, firstyear=700) # scenario: 'emission_bound' (without firm capacity)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When comparing the results of the original scenario without the reliability factor ('emission_bound') and the results of our newly modified scenario ('firm_capacity'), for the same carbon price we can observe that the activity of the wind and the coal powerplant is identical, which is what we hoped to achieve."
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Suggest to structure this as

## Activity
Description text
### Scenario 1

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 10]
```diff
@@ -196,6 +248,36 @@
     "p.plot_activity(baseyear=True, subset=['coal_ppl', 'wind_ppl'])"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The difference between the two scenarios can be seen when comparing the capacity.  Whereas in the 'emission_bound' scenario, coal capacity is quickly phased out, the 'firm_capacity' scenario shows, that there is a substantial amount of capacity required from the coal_ppl in order for the wind_ppl to achieve such high market share levels."
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Suggest to restructure as for *Activity* - it's odd when the explanation text and the section are not aligned.

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 11]
```diff
@@ -205,13 +287,34 @@
     "p.plot_capacity(baseyear=True, subset=['coal_ppl', 'wind_ppl'])"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This has a substantial impact on the electricity price, which is reflected in the prices for lighting."
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Suggest to restructure as above

File: tutorial/westeros/westeros_firm_capacity.ipynb [hunk 12]
```diff
@@ -1,5 +1,17 @@
 {
  "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Westeros Tutorial\n",
+    "## Adding representation of renewables - Part 1: Introducing `firm capacity`\n",
+    "\n",
+    "This tutorial, which demonstrates how to apply various model features to provide a more realistic representation of renewable energy integration in the nergy system, is comprised of three parts. In the first part we will show you how to ensure that in a model setup which does not use sub-annual timesteps and therefore does not depict peak load, we still account for `firm capacity`, i.e. sufficient backup capacity from electricity generation plants to maintain reliability through reasonable load and contingency events.\n",
```
Comment (danielhuppmann) [inline tutorial/westeros/westeros_firm_capacity.ipynb]: Typo `nergy`

**Comment by khaeru** [comment]
Today it was decided to schedule release 1.2.0 for 2019-04-01 in order to get #142 into the wild for use in teaching. I'm setting the milestone for this and the other tutorial update PRs (#145, #146, #147, #148) to this same milestone.

**Comment by gidden** [comment]
closing this for now since we decided to make one single PR with all tutorial additions/updates in combination with @behnam2015

**Pull request closed without merging**